core/src/Cabana_AoSoA.hpp

```diff
@@ -131,14 +132,14 @@ class AoSoA
     //! Member data types.
     using member_types = DataTypes;
 
-    //! Device type.
-    using device_type = DeviceType;
-
+    // FIXME: extracting the self type for backwards compatibility with previous
+    // template on DeviceType. Should simply be MemorySpace after next release.
     //! Memory space.
-    using memory_space = typename device_type::memory_space;
-
-    //! Execution space.
-    using execution_space = typename device_type::execution_space;
+    using memory_space = typename MemorySpace::memory_space;
```

Comment:
@dalg24 I wanted to check with you about this new change - without this we lose backwards compatibility with the previous `DeviceType` template because of the `size_type` below (and a few similar issues in other classes)

Reply:
So with this hack the change you propose support both a device type and a memory type?

Reply:
Do you want to attempt to emit a warning if someone is using a device type?

Reply:
Yes, since we've had device type in all of our examples up to now it seemed best to allow both. I think we should warn, but don't know a good way to do that

Reply:
```C++
constexpr bool warn_deprecated( std::false_type ) { return true; }
[[deprecated( "some helpful msg" )]] constexpr bool warn_deprecated( std::true_type ) { return true; }

template<...>
class AoSoA {
...
static_assert(warn_deprecated(Kokkos::is_device<MemorySpace>()));
};